### PR TITLE
changed architecture of USER

### PR DESCRIPTION
**Pull request by amit-levints-bgu**

### CODE REVIEW

File: src/main/java/Domain/UserController.java [hunk 1]
```diff
@@ -69,13 +58,22 @@ public boolean isAdmin(String userName) throws Exception {
...
+
+    public void addNewGuest(String id){
+        if (isGuestExists(id)) {
+            throw new IllegalArgumentException("Guest with ID " + id + " already exists.");
         }
-        throw new Exception("User not found.");
+        guestIds.add(id);
+    }
 
+    public void removeGuest(String id){
```
Comment (inbarbenchaim) [inline src/main/java/Domain/UserController.java:73]: when do you want to remove guest?

File: src/main/java/Domain/UserController.java [hunk 2]
```diff
@@ -69,13 +58,22 @@ public boolean isAdmin(String userName) throws Exception {
             throw new Exception("User not found.");
     }
 
-    // function that check if a given user is logged in
-    public boolean isLoggedIn(String userName) throws Exception {
-        User user = getUserByUsername(userName);
-        if (user != null) {
-            return user.isLoggedIn();
+    private boolean isGuestExists(String id) {
```
Comment (inbarbenchaim) [inline src/main/java/Domain/UserController.java:61]: when are you checking this?

File: src/main/java/ServiceLayer/UserService.java [hunk 3]
```diff
@@ -81,23 +88,22 @@ public Response isAdmin(String userId){
...
-            }
-        } catch (Exception e) {
-            response.setErrorMessage("Failed to check if user is logged in: " + e.getMessage());
-            logger.log(Level.SEVERE, "Failed to check if user is logged in: " + e.getMessage(), e);
-        }
-        return response;
-    }
-
-
+    //TODO: remove that function
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/UserService.java:91]: why didnt you removed? Metar?

File: src/main/java/Domain/UserController.java (hunk 2 above)
Comment (amit-levints-bgu) [reply src/main/java/Domain/UserController.java:61]: when adding/removing new guest

File: src/main/java/ServiceLayer/UserService.java (hunk 3 above)
Comment (inbarbenchaim) [reply src/main/java/ServiceLayer/UserService.java:91]: ok ask metar to do that pls after he is doing pull

**Pull request merged**